src/api/routers/participants/participantsCreation.ts

```diff
@@ -61,6 +71,30 @@ export async function validateParticipantCreationRequest(
   return errorMessage;
 }
 
+const createUserAndAssociatedParticipant = async (
+  parsedUser: z.infer<typeof UserCreationPartial>,
+  participantData: z.infer<typeof ParticipantCreationAndApprovalPartial>
+) => {
+  await User.transaction(async (trx) => {
+    // create user
+    const newPortalUser = await User.query(trx).insertAndFetch(parsedUser);
```

Comment:
`insertGraph` does not seem to support our use-case of the three-way mapping here, hence we are manually inserting the User, Participant and the User/Participant/Role mapping individually.